### DISCUSSION TITLE
Difference between Backup Type on Backup-DbaDatabase

### DISCUSSION THREAD

**Discussion by tomaustin700** [dataplat/dbatools#8271] [Q&A] [answered]
Hi,
Would someone be able to explain the difference between the backup type options for Backup-DbaDatabase please. They don't seem to directly map to SQL backup types so I just need a bit of clarification. `Full` and `Differential` make sense a they do directly map to SQL backup types but what does `Database` actually do? Also are `Differential` and `Diff` the same? Is `Log` Transaction Log or Tail Log? A little explanation of the different types and how they map to SQL backup types would be greatly appreciated.

Thanks

**Comment by CrazySwimmer** [comment, marked as answer]
I opened an issue (#8205) related to this some time ago. I am still waiting for feedback.

To answer your question, Database = Full, Differential = Diff, etc.
"Tail log backup" refers to a backup of the last part of the transaction log, before putting a database in a no-recovery or read-only state.

EDIT: Typos

**Reply by niphlod** [reply]
BTW, that's how SMOs enum are. 
You know about "Full", but in SMO's it has historically always been "Database".
You know about "Differential", but in SMO's it has historically always been "Database" + backuptype.Incremental = true.
I don't think anyone needs to explain that "Diff" is the same as "Differential" ;-) 
As for "Is Log Transaction Log or Tail Log ?" : Tail log backup is a transaction log backup either way, it's just the last one you do before performing a restore to be able to save the last bits of data you do not have backupped yet.
This is also known to you as "Tail Log", as it is in SSMS, but in TSQL is BACKUP ... WITH NORECOVERY and in SMOs it's Backup "Log" + backup.norecovery = true .... and in dbatools is mapped to `-NoRecovery`

I get you haven't a complete 1:1 mapping to what you're used to but as I just showed, not everyone calls the same things in the same way. Dbatools IMHO is a taddle bit friendlier than both SSMS, T-SQL or SMOs, while retaining the whole featureset.

**Reply by tomaustin700** [reply]
Thanks @CrazySwimmer and @niphlod, that really clears things up.
